File: NCS_LabiMichael/Quiz-App-MLV/Question-Service/src/main/java/com/ncs/question/service/QuestionService.java
package com.ncs.question.service;

import java.util.List;


import com.ncs.question.model.Question;


public interface QuestionService {

	public Question addQuestion(Question question);
	
	
	public List<Question> getAllQuestions();

	// Custom Repo
	public List<Question> getAllQuestionsByCategory(String category);

	public List<Question> getAllQuestionsByMarks(int marks);

	public List<Question> getAllQuestionsByCategoryAndMarks(String category, int marks);


	public Question updateAnswer(int questionId, String answer);

	public Question updateQuestion(Question question);
	
	public boolean deleteQuestion(int questionId);
	
	public List<Question> getQuiz(String category, int marks);
	
	
	

}

File: NCS_LabiMichael/QuizAppSPA/src/app/service/question.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Question } from '../model/question';

@Injectable({
  providedIn: 'root'
})
export class QuestionService {

  baseUrl: string =
  // 'http://localhost:8082/abc-university/admin/question';
  'http://localhost:8083/abc-university/question';

  constructor(private httpClient:HttpClient) { }

  getAllQuestions():
  Observable<Question[]>
  {
    console.log("Question Service - Get question called");
    return this.httpClient.get<Question[]>(this.baseUrl +'/allQuestions');
  }

  getQuiz(category:string, marks:number):
  Observable<Question[]>
  {

 
    console.log("Question Service - Get question called");
    return this.httpClient.get<Question[]>(this.baseUrl +'/getQuiz?/category='+category+"&marks="+marks);
  }

  getQuestionByCategory(category:string):
  Observable<Question[]>
  {
    console.log("Question Service - Get Question Called");
    return this.httpClient.get<Question[]>(this.baseUrl+'/question/getByCategory/'+category);
  }

  getQuestionByMarks(marks:number):
  Observable<Question[]>
  {
    console.log("Question Service - Get Question Called");
    return this.httpClient.get<Question[]>(this.baseUrl+'/question/getByMarks/'+marks);
  }

  addQuestion(question:Question):Observable<any>
  {
    return this.httpClient.post(this.baseUrl + '/add', question);
  }

  updateQuestion(question:Question):Observable<any>
  {
    return this.httpClient.put(this.baseUrl + '/update', question);
  }


  deleteQuestio(questionId:number):
Observable<any>
  {
return this.httpClient.delete(this.baseUrl+ '/question/'+ questionId);
  }


}//end of service

